ios/MaasCore/Sources/MaasComponents/Components/BottomSheet/BottomSheet.swift

```diff
@@ -0,0 +1,140 @@
+import SwiftUI
+import Combine
+
+struct BottomSheet: ViewModifier {
+
+    @Themeable(BottomSheetConstants.init) var constants
+
+    @EnvironmentObject var environment: BottomSheetEnvironment
+
+    @State var dragOffset: CGFloat = 0
+
+    @State var contentBounds: CGRect = .zero
+
+    @State var presenterBounds: CGRect = .zero
+
+    var topAnchor: CGFloat { presenterBounds.height - contentBounds.height }
+
+    var bottomAnchor: CGFloat { UIScreen.main.bounds.height }
+
+    var topInset: CGFloat {
+        UIApplication.shared.windows.first?.safeAreaInsets.top ?? 20.0
```

Comment:
20 is arbitrary?

Reply:
It's default status bar height if there are no **safeAreaInsets.top** 
<img width="339" alt="Screenshot 2021-02-15 at 11 27 10" src="https://user-images.githubusercontent.com/51905686/107928245-c0701500-6f80-11eb-8b6d-911e259d2385.png">